app/src/main/java/com/eveningoutpost/dexdrip/stats/StatsResult.java

```diff
@@ -75,14 +77,22 @@ public StatsResult(SharedPreferences settings, long from, long to){
             avg = 0;
         }
 
+        if(getTotalReadings() > 0){
```

Comment:
Would it be better to move this inside `getStdevUnitised()` so that the sql query is only run if that figure is requested? In `getTotal_carbs()` we calculate it the first time if requested and then store the result for efficiency if it is queried more than once.